src/pages/settings/SettingForm.tsx

```diff
@@ -19,33 +23,50 @@ export const getConfigId = (key: string) => {
 interface Props {
   configField: ConfigField;
   value?: string;
+  clusteredValue?: ClusterSpecificValues;
   isLast?: boolean;
 }
 
-const SettingForm: FC<Props> = ({ configField, value, isLast }) => {
+const SettingForm: FC<Props> = ({
+  configField,
+  value,
+  clusteredValue,
+  isLast,
+}) => {
   const { isRestricted } = useAuth();
   const [isEditMode, setEditMode] = useState(false);
   const notify = useNotify();
   const toastNotify = useToastNotification();
   const queryClient = useQueryClient();
   const { canEditServerConfiguration } = useServerEntitlements();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
 
   const editRef = useRef<HTMLDivElement | null>(null);
 
   // Special cases
   const isTrustPassword = configField.key === "core.trust_password";
   const isLokiAuthPassword = configField.key === "loki.auth.password";
+  const isMaasMachine = configField.key === "maas.machine";
   const isSecret = isTrustPassword || isLokiAuthPassword;
+  const isClusteredInput = isClustered && isMaasMachine;
 
   const settingLabel = (
     <ResourceLabel bold type="setting" value={configField.key} />
   );
 
-  const onSubmit = (newValue: string | boolean) => {
-    const config = {
-      [configField.key]: String(newValue),
-    };
-    updateSettings(config)
+  const onSubmit = (newValue: string | boolean | ClusterSpecificValues) => {
+    const isNotClustered =
+      typeof newValue !== "string" && typeof newValue !== "boolean";
```

Comment:
We can rely on `isClusteredInput` below directly. That variable is already defined above and will be more stable than defining a custom condition here based on the type of the value.